digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala

```diff
@@ -2169,11 +2165,25 @@ object DataFixture {
       val zippedFrozen = frozen.geometry.zip(frozen.geometry.tail)
 
       zippedFrozen.zipWithIndex.foldLeft(Seq.empty[RoadAddressTEMP]) {case (result, ((p1, p2), index)) =>
-        val addressF = geometryTransform.vkmGeometryTransform.coordToAddress(p1, includePedestrian = Some(true))
-        val addressL = geometryTransform.vkmGeometryTransform.coordToAddress(p2, includePedestrian = Some(true))
+
+      val (addressF, addressL ) = {
+          val addressOriginalF = geometryTransform.vkmGeometryTransform.coordToAddress(p1, includePedestrian = Some(true))
+          val addressOriginalL = geometryTransform.vkmGeometryTransform.coordToAddress(p2, includePedestrian = Some(true))
+
+          if (addressOriginalF.roadPart != addressOriginalL.roadPart) {
+            val geomAux = GeometryUtils.truncateGeometry2D(frozen.geometry, 0.05, GeometryUtils.geometryLength(frozen.geometry)-0.05)
+            val addrF = geometryTransform.vkmGeometryTransform.coordToAddress(geomAux.head, includePedestrian = Some(true))
+            val addrL = geometryTransform.vkmGeometryTransform.coordToAddress(geomAux.tail.head, includePedestrian = Some(true))
+
+            (addrF, addrL )
+
+          } else {
+            (addressOriginalF, addressOriginalL)
+          }
+      }
 
         val address = if(!(addressF.road == addressL.road && addressF.roadPart == addressL.roadPart)) {
-          val adj = roadLinkService.getAdjacent(frozen.linkId, false)
+          val adj = roadLinkService.getAdjacent(frozen.linkId, newTransaction = false)
```

Comment:
You don't need to specify the field name if they are by order, the previous code is fine like this one

Reply:
intelij suggestion.